### PR TITLE
Refactor commands and add testing

### PR DESCRIPTION
**Pull request by eguzki**
- [x] Refactor copy service command
- [x] Refactor update service command
- [x] Add unit tests

### CODE REVIEW

File: lib/3scale_toolbox/commands/copy_command/copy_service.rb [hunk 1]
```diff
@@ -27,178 +29,33 @@ def run
...
-              limit.delete('links')
-              client.create_application_plan_limit(copy_id, metrics_mapping.fetch(limit.fetch('metric_id')), limit)
-            end
-            puts "copied application plan #{copy_id} is missing #{missing_limits.size} from the original plan #{original_id}"
-          end
+          source_service = Entities::Service.new(id: arguments[:service_id],
+                                                 remote: remote(source))
+          copy_service = create_new_service(source_service.show_service, destination, system_name)
+          context = create_context(source_service, copy_service)
+          tasks = [Tasks::CopyServiceProxyTask.new(context),
```
Comment (mikz) [inline lib/3scale_toolbox/commands/copy_command/copy_service.rb]: Just a matter of style, but I think this is better for several reasons:

```ruby
tasks = [ 
  ...,
  ...,
]
```

Comment (mikz) [reply lib/3scale_toolbox/commands/copy_command/copy_service.rb]: Also, to enforce a uniform interface between them you can do:

```ruby
COPY_SERVICE = [
  Tasks::CopyServiceProxyTask,
  Tasks::CopyMethodsTas,
  ...,
].freeze
```

```ruby
COPY_SERVICE.each{ |task| task.new(context).call }
```

File: lib/3scale_toolbox/tasks/update_service_settings_task.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module ThreeScaleToolbox
+  module Tasks
+    class UpdateServiceSettingsTask
+      attr_reader :source_service, :copy_service, :target_system_name
+
+      def initialize(source_service:, copy_service:, target_name:)
```
Comment (mikz) [inline lib/3scale_toolbox/tasks/update_service_settings_task.rb]: When naming one `source` then the other should be `target` no?

File: lib/3scale_toolbox/tasks/copy_methods_task.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  module Tasks
+    class CopyMethodsTask
+      include CopyTask
+
+      def call
+        source_methods = source_service.methods
+        copy_methods = copy_service.methods
+        puts "original service hits metric #{source_service.hits['id']} has #{source_methods.size} methods"
+        puts "copied service hits metric #{copy_service.hits['id']} has #{copy_methods.size} methods"
+        m_m = ThreeScaleToolbox::Helper.array_difference(source_methods, copy_methods) do |method, copy|
```
Comment (mikz) [inline lib/3scale_toolbox/tasks/copy_methods_task.rb]: Writing `missing_methods` is not that big a deal.

**Comment by eguzki** [comment]
Opened another branch for integration tests. 

This PR is ready for review @mikz

**Pull request merged**